samples/extensions/khr/externalmemory/main.cpp

```diff
@@ -0,0 +1,559 @@
+        memcpy(vk_arr_y, arr_y.data(), sizeof(cl_float) * length);
+
+        // Get Vulkan external memory file descriptors for accessing external
+        // memory with OpenCL.
+        VkMemoryGetFdInfoKHR fd_info_x{};
+        fd_info_x.sType = VK_STRUCTURE_TYPE_MEMORY_GET_FD_INFO_KHR;
+        fd_info_x.pNext = nullptr;
+        fd_info_x.memory = vk_buf_x_memory;
+        fd_info_x.handleType = vk_external_memory_handle_type;
+        int fd_x;
+
+        VkMemoryGetFdInfoKHR fd_info_y{};
+        fd_info_y.sType = VK_STRUCTURE_TYPE_MEMORY_GET_FD_INFO_KHR;
+        fd_info_y.pNext = nullptr;
+        fd_info_y.memory = vk_buf_y_memory;
+        fd_info_y.handleType = vk_external_memory_handle_type;
+        int fd_y;
+
+        // We need to get the pointer to the vkGetMemoryFdKHR function because
+        // it's from extension VK_KHR_external_memory_fd.
+        PFN_vkGetMemoryFdKHR vkGetMemoryFdKHR =
+            (PFN_vkGetMemoryFdKHR)vkGetDeviceProcAddr(vk_device,
+                                                      "vkGetMemoryFdKHR");
+
+        VK_CHECK(vkGetMemoryFdKHR(vk_device, &fd_info_x, &fd_x));
+        VK_CHECK(vkGetMemoryFdKHR(vk_device, &fd_info_y, &fd_y));
+
+        // Create OpenCL buffers from Vulkan external memory file descriptors.
+        std::vector<cl_mem_properties> ext_mem_props_x = {
+            (cl_mem_properties)CL_EXTERNAL_MEMORY_HANDLE_OPAQUE_FD_KHR,
+            (cl_mem_properties)fd_x,
+            (cl_mem_properties)CL_DEVICE_HANDLE_LIST_KHR,
+            (cl_mem_properties)cl_device(),
+            CL_DEVICE_HANDLE_LIST_END_KHR,
+            0
+        };
+        std::vector<cl_mem_properties> ext_mem_props_y = {
+            (cl_mem_properties)CL_EXTERNAL_MEMORY_HANDLE_OPAQUE_FD_KHR,
+            (cl_mem_properties)fd_y,
+            (cl_mem_properties)CL_DEVICE_HANDLE_LIST_KHR,
+            (cl_mem_properties)cl_device(),
+            CL_DEVICE_HANDLE_LIST_END_KHR,
+            0
+        };
+
+        cl::Buffer cl_buf_x{ cl_context, ext_mem_props_x, CL_MEM_READ_ONLY,
+                             sizeof(cl_float) * length };
+        cl::Buffer cl_buf_y{ cl_context, ext_mem_props_y, CL_MEM_READ_WRITE,
+                             sizeof(cl_float) * length };
+
+        // Initialize queue for command execution.
+        cl_command_queue_properties queue_props[] = { CL_QUEUE_PROFILING_ENABLE,
+                                                      0 };
+        cl::CommandQueue queue{ cl_context, cl_device, *queue_props };
+
+        // Launch kernel.
+        if (diag_opts.verbose)
+        {
+            std::cout << "done.\nExecuting on device... ";
+            std::cout.flush();
+        }
+
+        std::vector<cl::Event> kernel_run;
+        auto dev_start = std::chrono::high_resolution_clock::now();
+        kernel_run.push_back(
+            saxpy(cl::EnqueueArgs{ queue, cl::NDRange{ length }, wgs }, a,
+                  cl_buf_x, cl_buf_y));
+        cl::WaitForEvents(kernel_run);
```

Comment:
One more - this is missing the calls to acquire and release the external memory objects.

See: https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_Ext.html#_acquiring_and_releasing_external_memory_objects

Reply:
Added!